### PR TITLE
Avoid calculating cos(arccos(cos(x))) in SolidAngle

### PR DESCRIPTION
**Pull request by quantumsteve**
**Description of work.**

This adds `V3D::cosAngle(const V3D &)` to more efficiently calculate this quantity. Added unit tests and Python export.

**To test:**

*There is no associated issue.*

*This does not require release notes* because implementation detail.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Kernel/test/V3DTest.h [hunk 1]
```diff
@@ -280,6 +280,17 @@ class V3DTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(a.angle(d), M_PI, 0.0001);
   }
 
+  void testCosAngle() {
+    a(2.0, 0.0, 0.0);
+    b(0.0, 1.0, 0.0);
+    c(1.0, 1.0, 0.0);
+    d(-1.0, 0.0, 0.0);
+    TS_ASSERT_DELTA(a.CosAngle(a), 1.0, 0.0001);
```
Comment (peterfpeterson) [inline Framework/Kernel/test/V3DTest.h]:
```suggestion
    TS_ASSERT_DELTA(a.cosAngle(a), 1.0, 0.0001);
```

**Review by peterfpeterson — DISMISSED** [review]
Reasonable addition.

**Comment by quantumsteve** [comment]
changing sqrt(a)*sqrt(b) to sqrt(a*b) broke a few tests :disappointed:

**Review by AndreiSavici — APPROVED** [review]
Unrelated build failure

**Pull request merged**